### PR TITLE
In 891 add linting mypy

### PR DESCRIPTION
**Pull request by ehanson8**
#### Helpful background context
Adding `mypy` and some miscellaneous changes. I will update the dependencies on the next PR adding `ruff`. There is much that should be refactored and I will address some of that in the next few PRs.

#### How can a reviewer manually see the effects of these changes?
As these are just linting updates, they can be verified by checking the CI workflow under the repo's `Actions` tab or cloning the repo and verifying that `make test` and `make lint` still pass.

#### What are the relevant tickets?
- https://mitlibraries.atlassian.net/browse/IN-891

#### Requires Database Migrations?
NO

#### Includes new or updated dependencies?
YES

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -1,3 +1,8 @@
+[mypy]
+
+[mypy-smart_open.*]
+ignore_missing_imports = True
```
Comment (ghukill) [inline setup.cfg:4]: Could be mistaken, but it seems to pass linting locally without this?

File: tests/test_cli.py [hunk 2]
```diff
@@ -1,11 +1,24 @@
...
 
 logger = logging.getLogger(__name__)
 
 
+def test_create_list_of_dspace_item_files():
+    file_list = create_list_of_dspace_item_files("111.1-111", {"key": "value"}, b"")
+    assert file_list == [
+        ("111.1-111.json", {"key": "value"}),
+        ("111.1-111.pdf", b""),
+    ]
```
[the remark this answers is not in the document]
Comment (ghukill) [reply tests/test_cli.py:19]: Sidebar: when adding `ruff`, I think we'll be able to have `mypy` annotation linting on the test suites as well, which would catch something like this.

**Review by ghukill — CHANGES_REQUESTED** [review]
Lots of work here, typing basically a whole project!  Nice work.

Per your PR notes, and discussions we've had, do see some areas where perhaps things could be refactored a bit, but know that's planned for future PRs.

In the context of adding type annotations, looks great to me, and will help dramatically during any refactoring.

File: awd/s3.py [hunk 3]
```diff
@@ -39,7 +45,7 @@ def archive_file_with_new_key(self, bucket, key, archived_key_prefix):
             Key=key,
         )
 
-    def put_file(self, file, bucket, key):
+    def put_file(self, file: str, bucket: str, key: str) -> PutObjectOutputTypeDef:
```
[the remark this answers is not in the document]
Comment (ehanson8) [reply awd/s3.py:48]: Using `str | bytes` which `ruff` will soon insist on 🙂

File: setup.cfg (hunk 1 above)
Comment (ehanson8) [reply setup.cfg:4]: I'm assuming this file is replaced by `pyproject.toml` in the next PR, can I leave until it's removed?

File: awd/cli.py [hunk 4]
```diff
@@ -199,15 +229,14 @@ def deposit(
...
-                for file in files_dict:
-                    s3_client.put_file(file["file_content"], bucket, file["file_name"])
+                for file_tuple in create_list_of_dspace_item_files(
+                    doi_file_name, json.dumps(metadata), wiley_response.content
+                ):
+                    s3_client.put_file(file_tuple[1], bucket, file_tuple[0])
             except ClientError as e:
                 logger.error(
-                    f"Upload failed: {file['file_name']},"
+                    f"Upload failed: {file_tuple[0]},"
```
[the remark this answers is not in the document]
Comment (ehanson8) [reply awd/cli.py:239]: That's so much better, great suggestion!

File: setup.cfg (hunk 1 above)
Comment (ghukill) [reply setup.cfg:4]: Good point!  Yep, think it will be removed.

File: awd/s3.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (ghukill) [reply awd/s3.py]: Ah, cool!  Love that syntax, didn't realize it was an option. FWIW, looks like python 3.10+.

**Review by ghukill — APPROVED** [review]
Looks good to me!

File: awd/s3.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (ehanson8) [reply awd/s3.py]: You are correct! [https://peps.python.org/pep-0604/]( https://peps.python.org/pep-0604/)

**Pull request merged**